OasisDEX/oasis-borrow
# Details section notification

# Details section notification

Support for stackable notifications for position details section.

![image](https://user-images.githubusercontent.com/16230404/219632932-435e3ee3-f8de-4a45-bd96-53861fa0d63a.png)
![image](https://user-images.githubusercontent.com/16230404/219633463-4a1324de-bb89-46be-b269-3d3492ea84c0.png)
  
## Changes 👷‍♀️

- Extended `DetailsSection` component with `notifications` props,
- created `DetailsSectionNotification` component.
  
## How to test 🧪

This is just a component, it's not used anywhere. If you want to test it, find any place with `Details section` and try to add one of the props listed below:

```typescript
        notifications={[
          {
            closable: true,
            icon: 'wallet',
            title: 'You are viewing 0x373...48304 vault',
            type: 'notice',
          },
        ]}
```
```typescript
        notifications={[
          {
            closable: true,
            icon: 'liquidation',
            title: 'Your vault got liquidated',
            message: 'A liquidation has happened on this vault. Reclaim 2.20 ETH',
            link: {
              label: 'Reclaim collateral',
              action: () => alert('Action clicked'),
            },
            type: 'error',
          },
        ]}
```
```typescript
        notifications={[
          {
            icon: 'coins_cross',
            title: 'This notification can\'t be closes and doesn\'t have any action.',
            type: 'warning',
          },
          {
            closable: true,
            icon: 'liquidation',
            title: 'This notification has link instead of action.',
            link: {
              label: 'Go to homepage',
              url: '/',
            },
            type: 'notice',
          },
        ]}
```
